services/supervisor/src/simple/mod.rs

```diff
@@ -69,6 +72,10 @@ impl SimpleSupervisorInterface for SimpleSupervisor {
             .verify_caller(Caller::as_transaction)
             .ok_or(DispatcherError::UnauthorizedCaller)?;
 
+        // Verify that transaction author is validator.
+        let author = context.caller().author().unwrap();
```

Comment:
`unwrap()` is really unsafe here ?

Reply:
Safe. We've checked `as_transaction` several lines above.